db/db_impl/db_impl_write.cc

```diff
@@ -92,15 +92,11 @@ Status DBImpl::MultiBatchWriteImpl(const WriteOptions& write_options,
       write_thread_.WaitForMemTableWriters();
     }
     WriteThread::WriteGroup wal_write_group;
-    mutex_.Lock();
-    bool need_log_sync = !write_options.disableWAL && write_options.sync;
-    bool need_log_dir_sync = need_log_sync && !log_dir_synced_;
+    LogContext log_context;
```

Comment:
why not init `need_log_dir_sync` here?

Reply:
`need_log_dir_sync ` was protected by `mutex_` before. This pr will protected this member variable by `log_write_mutex_` so it could only be accessed in the method `PreprocessWrite`.

Reply:
you mean `log_dir_synced_` is protected by `mutex_`?

Reply:
Yes.